### PR TITLE
AV-216533 : Remove double encoding for URI (#1522)

### PR DESCRIPTION
**Pull request by DixitAakash**
This PR removes double encoding for URI. Also adds two log statements. Similar changes have already been merged to release-1.12.2 branch as part of https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pull/1522.

### CODE REVIEW

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

**Comment by DixitAakash** [comment]
build ako

File: Makefile [hunk 1]
```diff
@@ -351,9 +351,16 @@ multitenancytests:
 	-v $(PWD):/go/src/$(PACKAGE_PATH_AKO) $(GO_IMG_TEST) \
 	$(GOTEST) -mod=vendor $(PACKAGE_PATH_AKO)/tests/multitenancytests/... -failfast -timeout 0 -coverprofile cover-21.out -coverpkg=./...
 
+.PHONY: namingtests
```
Comment (aaha97) [inline Makefile:354]: can we change the name to encoding tests or url tests.

**Comment by DixitAakash** [comment]
UT Run results :

```
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/k8stest	30.488s	coverage: 8.8% of statements in ./...

ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/integrationtest	499.077s	coverage: 26.1% of statements in ./...

ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/ingresstests	1012.967s	coverage: 38.8% of statements in ./...


sudo docker run \
-e ENDPOINTSLICES_ENABLED="false" \
-w=/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes \
-v /home/aviuser/ako-repo-fork/load-balancer-and-ingress-services-for-kubernetes:/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes golang:bullseye \
go test -v -mod=vendor github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/evhtests -failfast -timeout 0 -coverprofile cover-11.out -coverpkg=./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/evhtests	1686.362s	coverage: 38.1% of statements in ./...


sudo docker run \
-w=/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes \
-v /home/aviuser/ako-repo-fork/load-balancer-and-ingress-services-for-kubernetes:/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes golang:bullseye \
go test -v -mod=vendor github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/evhtests -failfast -timeout 0 -isVipPerNS=true -coverprofile cover-12.out -coverpkg=./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/evhtests	1591.795s	coverage: 37.5% of statements in ./...


sudo docker run \
-e ENDPOINTSLICES_ENABLED="false" \
-w=/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes \
-v /home/aviuser/ako-repo-fork/load-balancer-and-ingress-services-for-kubernetes:/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes golang:bullseye \
go test -v -mod=vendor github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/dedicatedevhtests -failfast -timeout 0 -coverprofile cover-13.out -coverpkg=./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/dedicatedevhtests	692.506s	coverage: 28.1% of statements in ./...



sudo docker run \
-w=/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes \
-v /home/aviuser/ako-repo-fork/load-balancer-and-ingress-services-for-kubernetes:/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes golang:bullseye \
go test -v -mod=vendor github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/dedicatedevhtests -failfast -timeout 0 -isVipPerNS=true -coverprofile cover-14.out -coverpkg=./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/dedicatedevhtests	788.482s	coverage: 28.4% of statements in ./...



ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/oshiftroutetests	1313.073s	coverage: 32.9% of statements in ./...


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/bootuptests	20.150s	coverage: 4.3% of statements in ./...


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/multicloudtests	170.448s	coverage: 8.2% of statements in ./...


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/advl4tests	46.714s	coverage: 19.0% of statements in ./...


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/namespacesynctests	152.391s	coverage: 21.1% of statements in ./...


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/servicesapitests	78.982s	coverage: 22.2% of statements in ./...


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/npltests	116.794s	coverage: 23.5% of statements in ./...


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/dedicatedvstests	187.448s	coverage: 29.4% of statements in ./...


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/hatests	33.830s	coverage: 24.9% of statements in ./...


sudo docker run \
-w=/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes \
-v /home/aviuser/ako-repo-fork/load-balancer-and-ingress-services-for-kubernetes:/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes golang:bullseye \
go test -v -mod=vendor github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests -failfast -cniPlugin=calico -coverprofile cover-18.out -coverpkg=./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests	103.532s	coverage: 13.0% of statements in ./...


sudo docker run \
-w=/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes \
-v /home/aviuser/ako-repo-fork/load-balancer-and-ingress-services-for-kubernetes:/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes golang:bullseye \
go test -v -mod=vendor github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests -failfast -cniPlugin=cilium -coverprofile cover-19.out -coverpkg=./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests	27.268s	coverage: 12.7% of statements in ./..


sudo docker run \
-u root:root \
-v /home/aviuser/ako-repo-fork/load-balancer-and-ingress-services-for-kubernetes/helm/ako:/apps \
-v /home/aviuser/ako-repo-fork/load-balancer-and-ingress-services-for-kubernetes/tests/helmtests:/apps/tests \
avi-buildops-docker-registry-02.eng.vmware.com:5000/avi-buildops/helmunittest/helm-unittest:3.11.1-0.3.0 .
PASS  Test avi-secret creation when avi controller credentials are specified or not specified in values.yaml	tests/avi-secret_test.yaml
PASS  Test AKO's ClusterRole with GatewayAPI	tests/clusterrole_test.yaml
PASS  Test GatewayClass creation with GatewayAPI	tests/gatewayclass_test.yaml
PASS  Test AKO's StatefulSet with GatewayAPI	tests/statefulset_test.yaml


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/infratests	0.925s


ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/urltests	15.232s	coverage: 22.0% of statements in ./...


	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests		coverage: 0.0% of statements
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests/graphlayer	288.913s	coverage: 18.9% of statements in ./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests/ingestion	270.605s	coverage: 3.0% of statements in ./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests/npltests	60.313s	coverage: 16.5% of statements in ./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests/status	314.697s	coverage: 16.0% of statements in ./...



```

File: Makefile (hunk 1 above)
Comment (DixitAakash) [reply Makefile]: @aaha97 I have changed it to urltests.

**Review by aaha97 — APPROVED** [review]
LGTM

**Review by akshayhavile — APPROVED** [review]

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Pull request merged**